TCAT/Controllers/RouteDetail+ContentViewController.swift

```diff
@@ -466,6 +515,76 @@ class RouteDetailContentViewController: UIViewController {
                 setIndex(of: marker, with: waypoint.wpType)
                 bounds = bounds.includingCoordinate(waypoint.coordinate)
             }
+            
+            if let busPath = path as? BusPath {
+                // Create circles at the first and last coordinate points / stops for bus route(s)
+                if let startBusStopCoordinate = busPath.waypoints.first {
+                    let startCircle = busStopCircles(at: startBusStopCoordinate.coordinate, on: mapView)
+                    finalDestinationCircles.append(startCircle)
+                }
+                
+                if let finalBusStopCoordinate = busPath.waypoints.last {
+                    let endCircle = busStopCircles(at: finalBusStopCoordinate.coordinate, on: mapView)
+                    finalDestinationCircles.append(endCircle)
+                }
+            }
+            
+            // Extract and append all coordinates of waypoints
+            if let walkPath = path as? WalkPath {
+                for circleInfo in walkPath.circles {
+                    let circle = GMSCircle(position: circleInfo.coordinate, radius: circleInfo.radius)
+                    if pathCount == 0 {
+                        firstRouteSegment.append(circle)
+                    } else {
+                        finalRouteSegment.append(circle)
+                    }
+                }
+            }
+            pathCount += 1
+        }
+        
+        func mapRouteSegment(_ segment: [GMSCircle], to path: GMSMutablePath, addMarker: Bool = false) {
+            for (index, waypoint) in segment.enumerated() {
```

Comment:
Nit: usually when we are not looping through a range or the indices of a list, we want to use a forEach loop instead of a for loop, it keeps the syntax more readable, so something like this:

As a rule of thumb, try to use a Swift higher order function where you can, rather than using a for or while loop

```
segment.enumerated().forEach { index, waypoint in
...
}
```